src/license_expression/__init__.py

```diff
@@ -1,1377 +1,1447 @@
-        if keyl in seen_keys:
-            dupe_keys.add(key)
-
-        # key cannot be an expression keyword
-        if keyl in KEYWORDS_STRINGS:
-            invalid_keys_as_kw.add(key)
-
-        # keep a set of unique seen keys
-        seen_keys.add(keyl)
-
-        # aliases is an optional attribute
-        aliases = getattr(symbol, 'aliases', [])
-        initial_alias_len = len(aliases)
-
-        # always normalize aliases for spaces and case
-        aliases = set([' '.join(alias.lower().strip().split()) for alias in aliases])
-
-        # KEEP UNIQUES, remove empties
-        aliases = set(a for a in aliases if a)
-
-        # issue a warning when there are duplicated or empty aliases
-        if len(aliases) != initial_alias_len:
-            warning_dupe_aliases.add(key)
-
-        # always add a lowercase key as an alias
-        aliases.add(keyl)
-
-        for alias in aliases:
-            # note that we do not treat as an error the presence of a duplicated
-            # alias pointing to the same key
-
-            # ensure that a possibly duplicated alias does not point to another key
-            aliased_key = seen_aliases.get(alias)
-            if aliased_key and aliased_key != keyl:
-                dupe_aliases[alias].append(key)
-
-            # an alias cannot be an expression keyword
-            if alias in KEYWORDS_STRINGS:
-                invalid_alias_as_kw[key].append(alias)
-
-            seen_aliases[alias] = keyl
-
-        if symbol.is_exception:
-            if keyl in seen_exceptions:
-                dupe_exceptions.add(keyl)
-            else:
-                seen_exceptions.add(keyl)
-
-    # build warning and error messages from invalid data
-    errors = []
-    for ind in sorted(not_symbol_classes):
-        errors.append('Invalid item: not a LicenseSymbol object: %(ind)s.' % locals())
-
-    for dupe in sorted(dupe_keys):
-        errors.append('Invalid duplicated license key: %(dupe)s.' % locals())
-
-    for dalias, dkeys in sorted(dupe_aliases.items()):
-        dkeys = ', '.join(dkeys)
-        errors.append('Invalid duplicated alias pointing to multiple keys: '
-                      '%(dalias)s point to keys: %(dkeys)s.' % locals())
-
-    for ikey, ialiases in sorted(invalid_alias_as_kw.items()):
-        ialiases = ', '.join(ialiases)
-        errors.append('Invalid aliases: an alias cannot be an expression keyword. '
-                      'key: "%(ikey)s", aliases: %(ialiases)s.' % locals())
-
-    for dupe in sorted(dupe_exceptions):
-        errors.append('Invalid duplicated license exception key: %(dupe)s.' % locals())
-
-    for ikw in sorted(invalid_keys_as_kw):
-        errors.append('Invalid key: a key cannot be an expression keyword: %(ikw)s.' % locals())
-
-    warnings = []
-    for dupeal in sorted(dupe_aliases):
-        errors.append('Duplicated or empty aliases ignored for license key: %(dupeal)r.' % locals())
-
-    return warnings, errors
+#

```

Comment:
I think that there is a problem with line endings in the whole file. We now have Windows CR/LF endings instead of the original POSIX LF

Reply:
I hope this (https://github.com/nexB/license-expression/commit/e84c64eca88c04e22f66461b5500077a96672e8c) fixed it.